src/components/FlightsTable/FlightsTable.jsx

```diff
@@ -7,9 +7,9 @@ export const FlightsTable = ({originFilter, flights}) => {
     <section className="flightsTable">
       <h3>Flights</h3>
       <section>
-        {flights.filter((flight) => (
+        {flights?.filter((flight) => (
```

Comment:
Filtering in-place in your JSX is bad practice. The JSX tends to be messy enough without any sort of JS logic in it.

A cleaner strategy would be to have some sort of "filteredFlights" `useState` that is updated as a side-effect of `flights` and `originFilter`. Then you would just need a single map statement in your JSX.

Reply:
That makes a lot of sense. I removed the filter code from the FlightTable jsx and added this use effect to my App component:

```
  useEffect(() => {
      if(!originFilter.length) {
        setFilteredFlights(flights);
      } else {
        setFilteredFlights(flights.filter((flight) => originFilter.includes(flight.origin)));
      }
  }, [originFilter])
```

Is there a best practice in regards to ensuring there are not too many useState/useEffect/data manipulation functions in the main App component?

Reply:
Generally you want to use `useEffect()` only for dependencies that are external to your component (such as calling an API) that you need to sync with your state.

Since changing state inside the component causes the component to re-render, there's no need for a `useEffect` to update something internally. The React docs state "You don't Effects to transform data for rendering." (See the doc "You might not need an Effect" here: https://react.dev/learn/you-might-not-need-an-effect)

So instead of having `const [filteredFlights, setFilteredFlights] = useState(......)` and then updating that with `useEffect`, you can just do this:

```
// ... some stuff above this
const [flights, setFlights] = useState(null);
const [originFilter, setOriginFilter] = useState([]);
// ... some stuff below this

let filteredFlights = flights;
if (originalFilter.length) {
    filteredFlights = flights.filter((flight) => originFilter.includes(flight.origin));
}

// ... rest of the component
```

Reply:
> Is there a best practice in regards to ensuring there are not too many useState/useEffect/data manipulation functions in the main App component?
 
Nothing specific but follow the typical guidelines for any file of code or function -- probably shouldn't have more than a few hundred lines of code. You can always split useEffect() etc. out into their own custom hooks and import those into the component.

But as mentioned read through this first:  https://react.dev/learn/you-might-not-need-an-effect. Not only is it good guidelines, it provides further insight into how state management, hooks, and component rendering really works.

Reply:
Again, super helpful. I didn't put together that since the setOriginFilter would intrinsically re-render the component, I could just have logic baked into the App component outside of a hook that would run and update a variable value. I've fixed this.